### ISSUE TITLE
obsolete hyperkit driver: Error in driver during machine creation: vmnet: error from vmnet.framework: -1

### ISSUE THREAD

**Issue by matti** [kubernetes/minikube#4454]
This issue is ~duplicate, but since those other issues are closed, I'm opening a new one that has consistent repro instructions.

minikube version: v1.1.1
hyperkit: v0.20180403-17-g3e954c
osx high sierra

```
minikube start -p makkara-1 --vm-driver hyperkit --uuid 00000000-0000-0000-0000-000000000001 --cpus 12 --memory 2048
# ok
minikube delete -p makkara-1
# ok 
```

```
minikube start -p makkara-1 --vm-driver hyperkit --uuid 00000000-0000-0000-0000-000000000001 --cpus 12 --memory 2048
😄  minikube v1.1.1 on darwin (amd64)
🔥  Creating hyperkit VM (CPUs=12, Memory=2048MB, Disk=20000MB) ...
E0609 12:42:07.316375   60161 start.go:529] StartHost: create: Error creating machine: Error in driver during machine creation: vmnet: error from vmnet.framework: -1

💣  Unable to start VM: create: Error creating machine: Error in driver during machine creation: vmnet: error from vmnet.framework: -1
```

then after 10s running again it starts, but won't start (hangs):
```
minikube -v 100 start -p makkara-1 --vm-driver hyperkit --uuid 00000000-0000-0000-0000-000000000001 --cpus 12 --memory 2048
😄  minikube v1.1.1 on darwin (amd64)
Found binary path at /usr/local/bin/docker-machine-driver-hyperkit
Launching plugin server for driver hyperkit
Plugin server listening at address 127.0.0.1:59656
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(makkara-1) Calling .DriverName
(makkara-1) Calling .GetState
(makkara-1) Calling .DriverName
🔄  Restarting existing hyperkit VM for "makkara-1" ...
(makkara-1) Calling .Start
(makkara-1) Using UUID 00000000-0000-0000-0000-000000000001
(makkara-1) Generated MAC ce:40:4d:c8:6:17
(makkara-1) Starting with cmdline: loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 systemd.legacy_systemd_cgroup_controller=yes base host=makkara-1
(makkara-1) Calling .GetConfigRaw
⌛  Waiting for SSH access ...
(makkara-1) Calling .DriverName
Waiting for SSH to be available...
Getting to WaitForSSH function...
(makkara-1) Calling .GetSSHHostname
(makkara-1) Calling .GetSSHPort
(makkara-1) Calling .GetSSHKeyPath
(makkara-1) Calling .GetSSHKeyPath
(makkara-1) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x14813c0] 0x1481390 <nil>  [] 0s} 192.168.64.6 22 <nil> <nil>}
About to run SSH command:
exit 0
```

and now `ps grep aux | grep makkara-1` does not show `hyperkit` running.

However, upgrading `docker-machine-driver-hyperkit`:
```
curl -LO https://storage.googleapis.com/minikube/releases/latest/docker-machine-driver-hyperkit && sudo install -o root -g wheel -m 4755 docker-machine-driver-hyperkit /usr/local/bin/
```

as suggested in https://github.com/kubernetes/minikube/issues/2994#issuecomment-495007400 fixes this.

**Comment by yinzara** [comment]
Unfortunately upgrading to the most recent driver also does not solve this issue.  I have discovered this issue is somehow linked to having a USB networking adapter attached to the PC. If I disconnect my USB network adapter, delete it from System Settings, then start Minikube, the above changes to the following:

```
...
⌛  Waiting for SSH access ...
(minikube) Calling .DriverName
Waiting for SSH to be available...
Getting to WaitForSSH function...
(minikube) Calling .GetSSHHostname
(minikube) Calling .GetSSHPort
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x14813c0] 0x1481390 <nil>  [] 0s} 192.168.64.5 22 <nil> <nil>}
About to run SSH command:
exit 0
(minikube) DBG | 2019/06/17 10:53:08 fcntl(F_PUNCHHOLE) failed: host filesystem does not support sparse files: Operation not permitted
(minikube) DBG | 2019/06/17 10:53:08 rdmsr to register 0x140 on vcpu 0
(minikube) DBG | 2019/06/17 10:53:09 rdmsr to register 0x140 on vcpu 1
(minikube) DBG | 2019/06/17 10:53:09 rdmsr to register 0x140 on vcpu 2
(minikube) DBG | 2019/06/17 10:53:09 rdmsr to register 0x140 on vcpu 3
(minikube) DBG | 2019/06/17 10:53:22 rdmsr to register 0x64e on vcpu 3
(minikube) DBG | 2019/06/17 10:53:22 rdmsr to register 0x34 on vcpu 3
SSH cmd err, output: <nil>: 
Detecting the provisioner...
(minikube) Calling .GetSSHHostname
(minikube) Calling .GetSSHPort
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x14813c0] 0x1481390 <nil>  [] 0s} 192.168.64.5 22 <nil> <nil>}
About to run SSH command:
cat /etc/os-release
SSH cmd err, output: <nil>: NAME=Buildroot
VERSION=2018.05
ID=buildroot
VERSION_ID=2018.05
PRETTY_NAME="Buildroot 2018.05"

found compatible host: buildroot
(minikube) Calling .GetMachineName
setting hostname "minikube"
(minikube) Calling .GetMachineName
(minikube) Calling .GetSSHHostname
(minikube) Calling .GetSSHPort
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHUsername
Using SSH client type: native
&{{{<nil> 0 [] [] []} docker [0x14813c0] 0x1481390 <nil>  [] 0s} 192.168.64.5 22 <nil> <nil>}
About to run SSH command:
sudo hostname minikube && echo "minikube" | sudo tee /etc/hostname
I0617 10:53:43.502391   28816 cache_images.go:302] 
I0617 10:53:43.502394   28816 cache_images.go:302] 
I0617 10:53:43.502422   28816 cache_images.go:302] 
I0617 10:53:43.502462   28816 cache_images.go:302] 
I0617 10:53:43.502428   28816 cache_images.go:302] 
I0617 10:53:43.502436   28816 cache_images.go:302] 
I0617 10:53:43.502483   28816 cache_images.go:302] 
I0617 10:53:43.502465   28816 cache_images.go:302] 
I0617 10:53:43.502475   28816 cache_images.go:302] 
I0617 10:53:43.502478   28816 cache_images.go:302] 
I0617 10:53:43.502567   28816 cache_images.go:302] 
I0617 10:53:43.502567   28816 cache_images.go:302] 
I0617 10:53:43.502623   28816 cache_images.go:302] 
I0617 10:53:43.502634   28816 cache_images.go:85] Successfully cached all images.
SSH cmd err, output: <nil>: minikube

(minikube) Calling .GetSSHHostname
(minikube) Calling .GetSSHPort
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHKeyPath
(minikube) Calling .GetSSHUsername
Using SSH client type: native
...
```

**Comment by tstromberg** [comment]
Thanks for the report! We're introducing a check in minikube to detect obsolete hyperkit drivers. I'll leave this issue open until then.

@yinzara - your issue appears unrelated. Please open a new issue if possible.

**Comment by tstromberg** [comment]
This check landed in v1.3.1.
